### PR TITLE
feat: enhance newWindow function to support 'tab' or 'window' types

### PR DESCRIPTION
**Pull request by harsha509**
## Proposed changes

Fixes: https://github.com/webdriverio/webdriverio/issues/13790
- Added 'type' option to newWindow function to specify 'tab' or 'window' (default 'window')
- Updated return value to include both window handle and type
- Improved JSDoc to reflect new behaviour and clarify usage, including combined @throws clause
- Added tests to validate window and tab types

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [x] This change is solely for `v9` and doesn't need to be back-ported
- [ ] Back-ported PR at `#XXXXX`

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/webdriverio/tests/commands/browser/newWindow.test.ts [hunk 1]
```diff
@@ -108,4 +108,36 @@ describe('newWindow', () => {
...
+            }
+        })
+
+        const newHandle = await browser.newWindow('https://webdriver.io', {
+            type: 'tab',
+            windowName: 'some tab'
+        })
+
+        expect(newHandle.type).toBe('tab')
+    })
```
Comment (christian-bromann) [inline packages/webdriverio/tests/commands/browser/newWindow.test.ts:141]: Can we update these and actually check if the bidi command is called correctly? In order to run a mocked Bidi session the `browserName` needs to be called `bidi` and we have to spy on some commands on the `browser` object. For inspiration, please take a look at the `switchTab` command.

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 2]
```diff
@@ -8,54 +8,86 @@ const WAIT_FOR_NEW_HANDLE_TIMEOUT = 3000
...
+          await browser.switchToWindow(handles[0])
+          console.log(await browser.getTitle()) // outputs: "Google"
+     });
+ * </example>
  *
  * @param {string}  url      website URL to open
  * @param {NewWindowOptions=} options                newWindow command options
+ * @param {string=}           options.type         type of new window: 'tab' or 'window'
  * @param {String=}           options.windowName     name of the new window
  * @param {String=}           options.windowFeatures features of opened window (e.g. size, position, scrollbars, etc.)
```
[the remark this answers is not in the document]
Comment (harsha509) [reply packages/webdriverio/src/commands/browser/newWindow.ts:60]: HI Chris,
Thanks for the review. You're right, `windowFeatures` don't apply as expected with newWindow in classic mode (and we're not passing windowName or windowFeatures in BiDi mode). I've added a log warning for this.

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 3]
```diff
@@ -67,7 +99,7 @@ export async function newWindow (
 
     if (this.isBidi) {
         const contextManager = getContextManager(this)
-        const { context } = await this.browsingContextCreate({ type: 'window' })
+        const { context } = await this.browsingContextCreate({ type: type })
         contextManager.setCurrentContext(context)
```
[the remark this answers is not in the document]
Comment (harsha509) [reply packages/webdriverio/src/commands/browser/newWindow.ts:103]: I tried accessing elements, and it seems to be working as expected. I believe the context is set correctly when not in BiDi mode in here

https://github.com/webdriverio/webdriverio/blob/c2b4e381e24aefff844317c3895d47d2c0c145ad/packages/webdriverio/src/commands/browser/newWindow.ts#L114-L129

File: packages/webdriverio/tests/commands/browser/newWindow.test.ts (hunk 1 above)
Comment (harsha509) [reply packages/webdriverio/tests/commands/browser/newWindow.test.ts]: Added tests as suggested!

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 4]
```diff
@@ -66,8 +104,11 @@ export async function newWindow (
     const tabsBefore = await this.getWindowHandles()
 
     if (this.isBidi) {
+        if (windowName || windowFeatures) {
+            log.warn('The "windowName" and "windowFeatures" options are not supported in BiDi mode and will be ignored.')
+        }
         const contextManager = getContextManager(this)
-        const { context } = await this.browsingContextCreate({ type: 'window' })
+        const { context } = await this.browsingContextCreate({ type: type })
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/newWindow.ts:111]:
```suggestion
        const { context } = await this.browsingContextCreate({ type })
```

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 5]
```diff
@@ -3,59 +3,97 @@ import { sleep } from '@wdio/utils'
...
+    /**
+    * Validate the 'type' parameter to ensure it is either 'tab' or 'window'
+    */
+    if (!['tab', 'window'].includes(type)) {
+        throw new Error(`Invalid type '${type}' provided to newWindow command. Use either 'tab' or 'window'`)
+    }
+
+    if (windowFeatures) {
+        log.warn('The "windowFeatures" option is deprecated and will be removed in future versions.')
+    }
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/newWindow.ts:95]: We should just generally deprecate this

```suggestion
    if (windowName || windowFeatures) {
        log.warn('The "windowName" and "windowFeatures" options are deprecated and only supported in WebDriver Classic sessions.')
    }
```

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 6]
```diff
@@ -66,8 +104,11 @@ export async function newWindow (
     const tabsBefore = await this.getWindowHandles()
 
     if (this.isBidi) {
+        if (windowName || windowFeatures) {
+            log.warn('The "windowName" and "windowFeatures" options are not supported in BiDi mode and will be ignored.')
+        }
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/newWindow.ts:109]:
```suggestion
```

**Review by christian-bromann — COMMENTED** [review]
Some minor remaining suggestions

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 7]
```diff
@@ -3,59 +3,97 @@ import { sleep } from '@wdio/utils'
...
         console.log(await browser.getTitle()) // outputs: "Google"
     });
  * </example>
+ * <example>
+      :newTabSync.js
+      it('should open a new tab', async () => {
+          await browser.url('https://google.com')
+            console.log(await browser.getTitle()) // outputs: "Google"
+
+         await browser.newWindow('https://webdriver.io', {
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/newWindow.ts:43]:
```suggestion
          await browser.newWindow('https://webdriver.io', {
```

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 8]
```diff
@@ -3,59 +3,97 @@ import { sleep } from '@wdio/utils'
...
+          await browser.switchToWindow(handles[1])
+          await browser.closeWindow()
+          await browser.switchToWindow(handles[0])
+          console.log(await browser.getTitle()) // outputs: "Google"
+     });
+ * </example>
  *
  * @param {string}  url      website URL to open
  * @param {NewWindowOptions=} options                newWindow command options
+ * @param {string=}           options.type         type of new window: 'tab' or 'window'
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/newWindow.ts:60]:
```suggestion
 * @param {string=}           options.type           type of new window: 'tab' or 'window'
```

**Review by christian-bromann — APPROVED** [review]
Minor space adjustments, rest LGTM 👍

File: packages/webdriverio/src/commands/browser/newWindow.ts [hunk 9]
```diff
@@ -3,59 +3,97 @@ import { sleep } from '@wdio/utils'
...
         await browser.closeWindow()
         await browser.switchToWindow(handles[0])
         console.log(await browser.getTitle()) // outputs: "Google"
     });
  * </example>
+ * <example>
+      :newTabSync.js
+      it('should open a new tab', async () => {
+          await browser.url('https://google.com')
+            console.log(await browser.getTitle()) // outputs: "Google"
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/newWindow.ts:41]:
```suggestion
          console.log(await browser.getTitle()) // outputs: "Google"
```

**Pull request merged**